src/anomalib/utils/benchmarking/benchmark.py

```diff
@@ -0,0 +1,315 @@
+
+    Returns:
+        dict: Collection of all the metrics such as time taken, throughput and performance scores.
+    """
+
+    with TemporaryDirectory() as project_path:
+        model_config.results_dir.path = project_path
+        datamodule = get_datamodule(model_config)
+        model = get_model(model_config)
+
+        callbacks = get_sweep_callbacks(model_config)
+
+        engine = Engine(**model_config.trainer, logger=None, callbacks=callbacks)
+
+        start_time = time.time()
+
+        engine.fit(model=model, datamodule=datamodule)
+
+        # get start time
+        training_time = time.time() - start_time
+
+        # Creating new variable is faster according to https://stackoverflow.com/a/4330829
+        start_time = time.time()
+        # get test results
+        test_results = engine.test(model=model, datamodule=datamodule)
+
+        # get testing time
+        testing_time = time.time() - start_time
+
+        # Create dirs for torch export (as default only lighting model is produced)
+        export(
+            task=model_config.dataset.task,
+            transform=engine.trainer.datamodule.test_data.transform.to_dict(),
+            input_size=model_config.model.input_size,
+            model=model,
+            export_mode=ExportMode.TORCH,
+            export_root=project_path,
+        )
+
+        throughput = get_torch_throughput(
+            model_path=project_path,
+            test_dataset=datamodule.test_dataloader().dataset,
+            device=model_config.trainer.accelerator,
+        )
+
+        # Get OpenVINO metrics
+        openvino_throughput = float("nan")
+        if openvino_metrics:
+            # Create dirs for openvino model export
+            export(
+                task=model_config.dataset.task,
+                transform=engine.trainer.datamodule.test_data.transform.to_dict(),
+                input_size=model_config.model.input_size,
+                model=model,
+                export_mode=ExportMode.OPENVINO,
+                export_root=project_path,
+            )
+            openvino_throughput = get_openvino_throughput(model_path=project_path, test_dataset=datamodule.test_data)
+
+        # arrange the data
+        data = {
+            "Training Time (s)": training_time,
+            "Testing Time (s)": testing_time,
+            f"Inference Throughput {model_config.trainer.accelerator} (fps)": throughput,
+            "OpenVINO Inference Throughput (fps)": openvino_throughput,
+        }
+        for key, val in test_results[0].items():
+            data[key] = float(val)
+
+    return data
+
+
+def compute_on_cpu(sweep_config: DictConfig | ListConfig, folder: str | None = None):
+    """Compute all run configurations over a sigle CPU."""
+    for run_config in get_run_config(sweep_config.grid_search):
+        model_metrics = sweep(run_config, 0, sweep_config.seed_everything, False)
```

Comment:
to make it clear what each argument represents, maybe we could add keyword args here

```suggestion
        model_metrics = sweep(
            run_config=run_config,
            device=0,
            seed=sweep_config.seed_everything,
            convert_openvino=False
        )
```